api/src/opentrons/protocol_engine/commands/command.py

```diff
@@ -1,100 +1,92 @@
+    """Base class for command creation requests.
 
+    You shouldn't use this class directly; instead, use or define
+    your own subclass per specific command type.
+    """
 
-@dataclass(frozen=True)
-class RunningCommand(Generic[ReqT, ResT]):
-    """A command that is currently being executed."""
+    commandType: str = Field(
+        ...,
+        description=(
+            "Specific command type that determines data requirements and "
+            "execution behavior"
+        ),
+    )
+    data: CommandDataT = Field(..., description="Command execution data payload")
 
-    created_at: datetime
-    started_at: datetime
-    request: ReqT
 
-    def to_completed(
-        self,
-        result: ResT,
-        completed_at: datetime,
-    ) -> CompletedCommand[ReqT, ResT]:
-        """Create a CompletedCommand from a RunningCommand."""
-        return CompletedCommand(
-            created_at=self.created_at,
-            started_at=self.started_at,
-            request=self.request,
-            completed_at=completed_at,
-            result=result
-        )
-
-    def to_failed(
-        self,
-        error: ProtocolEngineError,
-        failed_at: datetime,
-    ) -> FailedCommand[ReqT]:
-        """Create a FailedCommand from a RunningCommand."""
-        return FailedCommand(
-            created_at=self.created_at,
-            started_at=self.started_at,
-            request=self.request,
-            failed_at=failed_at,
-            error=error
-        )
-
-
-@dataclass(frozen=True)
-class PendingCommand(Generic[ReqT, ResT]):
-    """A command that has not yet been started."""
-
-    created_at: datetime
-    request: ReqT
-
-    def to_running(self, started_at: datetime) -> RunningCommand[ReqT, ResT]:
-        """Create a RunningCommand from a PendingCommand."""
-        return RunningCommand(
-            created_at=self.created_at,
-            request=self.request,
-            started_at=started_at,
-        )
-
-
-class CommandImplementation(ABC, Generic[ReqT, ResT]):
-    """Abstract command implementation.
+class BaseCommand(GenericModel, Generic[CommandDataT, CommandResultT]):
+    """Base command model.
+
+    You shouldn't use this class directly; instead, use or define
+    your own subclass per specific command type.
+    """
+
+    id: str = Field(..., description="Unique identifier for a particular command")
+    createdAt: datetime = Field(..., description="Command creation timestamp")
+    commandType: str = Field(
+        ...,
+        description=(
+            "Specific command type that determines data requirements and "
+            "execution behavior"
+        ),
+    )
+    status: CommandStatus = Field(..., description="Command execution status")
+    data: CommandDataT = Field(..., description="Command execution data payload")
+    result: Optional[CommandResultT] = Field(
+        None,
+        description="Command execution result data, if completed",
+    )
```

Comment:
```suggestion
    result: Optional[CommandResultT] = Field(
        None,
        description="Command execution result data, if succeeded",
    )
```
Very nitty nitpick for disambiguation. (Did a "failed" command "complete"?)